tests/structured-data/archive-post-test.php

```diff
@@ -99,17 +99,79 @@ public function test_blog_page_for_posts_paged() {
 		self::assertEquals( get_permalink( $page_id ) . 'page/2', $structured_data['url'] );
 	}
 
+	public function test_author_archive() {
+		// Set permalinks, as Parsely currently strips ?page_id=... from the URL property.
+		// See https://github.com/Parsely/wp-parsely/issues/151
+		$this->set_permalink_structure( '/%postname%/' );
+
+		// Setup Parsley object.
+		$parsely         = new \Parsely();
+		$parsely_options = get_option( \Parsely::OPTIONS_KEY );
+
+		// Insert a single user, and a Post assigned to them.
+		$user = self::factory()->user->create( [ 'user_login' => 'parsely' ] );
```

Comment:
Is there a reason this is using a static method call instead of `$this->factory`?  Seems like other usages are all the latter.  There's a few other places in this change I see `self::factory()` as well.

Reply:
`self::factory()->...` is what WP core uses. See https://core.trac.wordpress.org/browser/trunk/tests/phpunit/tests/query/taxQuery.php

Just like `$this->assert...` and `self::assert...` (see "A comment question" paragraph at the top of https://phpunit.readthedocs.io/en/9.5/assertions.html) there doesn't seem to be any difference - but to me, it helps signify that it's a utility method being called, rather than a protected / private method in the test class itself.

Reply:
Besides - the actual `factory()` method on the WP unit test case, is static: https://core.trac.wordpress.org/browser/trunk/tests/phpunit/includes/abstract-testcase.php#L41

Reply:
Cool.  Definitely makes sense to me to standardize on the static syntax.